pkg/summarizer/flakiness.go

```diff
@@ -0,0 +1,286 @@
+package summarizer
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/testgrid/internal/result"
+	"github.com/GoogleCloudPlatform/testgrid/pb/state"
+	"github.com/sirupsen/logrus"
+	"vbom.ml/util/sortorder"
+)
+
+var INFRA_REGEX = regexp.MustCompile(`^\w+$`)
+var JAILED_REGEX = regexp.MustCompile(`(JAILED )?(\/?\/?[A-Za-z0-9\/_\-.:}{]+) - \[([A-Za-z0-9\/:\-_.\s]+)\]\s*`)
+
+// Keep in mind that flakiness is measured as out of 100, i.e. 23 not .23
+var DEFAULT_FLAKINESS = 50.0
+var MIN_RUNS = 0
+var HEALTHY_RANGE = map[int]string{
+	20: "red",
+	10: "purple",
+	3:  "orange",
+	0:  "green",
+}
+
+type Result struct {
+	name             string
+	passed           int
+	failed           int
+	flakyCount       int
+	averageFlakiness float64
+	failedInfraCount int
+	infraFailures    map[string]int
+}
+
+// Temporary structs while I decide what will go into summary.proto
+type Healthiness struct {
+	startDate        int
+	endDate          int
+	tests            []TestInfo
+	totalTests       int
+	totalJailedTests int
+	averageFlakiness float64
+	flakyBuckets     []FlakyBucket
+	infraIssues      map[string]int
+	totalConfigs     int
+}
+
+type TestInfo struct {
+	name               string
+	env                string
+	totalRuns          int
+	totalRunsWithInfra int
+	passedRuns         int
+	failedRuns         int
+	failedInfraRuns    int
+	flakyRuns          int
+	flakiness          float64
+	infraInfo          string
+}
+
+type FlakyBucket struct {
+	threshold float64
+	tests     int
+}
+
+// For sorting, primarily intended for map[string]int as implemented below
+type IntString struct {
+	i int
```

Comment:
Nit: Maybe flip these so it is in Key-Value order

Reply:
Done; though I initially put them in Value-Key because that was the sort order and it made the most sense in my head, but I can see how it would be confusing for others.